docs/source/dev/ci.rst

```diff
@@ -13,9 +13,119 @@ GitLab CI
 
 We use ``GitLab CI`` because it allows us to use self-hosted system, e.g. GPU systems. The GitHub repository is mirrored on https://gitlab.com/hzdr/crp/alpaka . Every commit or pull request is automatically mirrored to GitLab and triggers the CI. The configuration of the ``GitLab CI`` is stored in the file ``.gitlab-ci.yml``. The workflow of a ``GitLab CI`` is different from ``GitHub Actions``. Instead of downloading an unmodified container from ``Docker Hub`` and preparing the environment during the test job, ``GitLab CI`` uses containers which are already prepared for the tests. The containers are built in an `extra repository <https://gitlab.hzdr.de/crp/alpaka-group-container>`_ and contain all dependencies for alpaka. All available containers can be found `here <https://gitlab.hzdr.de/crp/alpaka-group-container/container_registry>`_. The scripts to build alpaka and run the tests are shared with ``GitHub Actions`` and located at ``script/``.
 
+Most of the jobs for the GitLab CI are generated automatically. For more information, see the section :ref:`The Job Generator`.
+
+It is also possible to define custom jobs, see :ref:`Custom jobs`.
+
 .. figure:: /images/arch_gitlab_mirror.svg
    :alt: Relationship between GitHub.com, GitLab.com and HZDR gitlab-ci runners
 
    Relationship between GitHub.com, GitLab.com and HZDR gitlab-ci runners
 
-To change how the tests are built and executed, modify the code in the `alpaka repository <https://github.com/alpaka-group/alpaka>`_. If the container environment with the dependencies needs to be changed, please open an issue or contribute to `alpaka-group-container repository <https://gitlab.hzdr.de/crp/alpaka-group-container>`_.
+The Container Registry
+++++++++++++++++++++++
+
+Alpaka uses containers in which as many dependencies as possible are already installed to save job execution time. The available containers can be found `here <https://gitlab.hzdr.de/crp/alpaka-group-container/container_registry>`_. Each container provides a tool called ``agc-manager`` to check if a software is installed. The documentation for ``agc-manager`` can be found `here <https://gitlab.hzdr.de/crp/alpaka-group-container/-/tree/master/tools>`_. A common way to check if a software is already installed is to use an ``if else statement``. If a software is not installed yet, you can install it every time at job runtime.
```

Comment:
Personally, I prefer to keep lines shorter also in documentation files. For example, I add a newline after each sentence.